### DISCUSSION TITLE
Compartilhar variáveis entre pipes no gulp

### DISCUSSION THREAD

**Discussion by yuricamara** [frontendbr/forum#125] [Perguntas e Respostas]
Quem manja de compartilhar variáveis entre pipes? Usando uma variável no escopo da função da task não funciona!

Trecho do código:

``` javascript
.pipe(plugins.intercept(function(file){
  var content = JSON.parse(file.contents.toString());

  cssTimestampStored = content.css

  return file;
}))
.pipe(plugins.rename('ionic.app.' + cssTimestampStored + '.min.css'))
```

No caso, a variável em questão é a  cssTimestampStored

**Comment by tcelestino** [comment]
@yuricamara ver se isso te ajuda - http://stackoverflow.com/questions/22026389/passing-variables-between-pipes-in-gulp

**Comment by yuricamara** [comment]
@tcelestino Obrigado pelo link. Eu tentei as sugestões dele sem sucesso. Acabei resolvendo meu problema de outra forma mas continuo querendo aprender a fazer isso.

**Comment by tcelestino** [comment]
@yuricamara qual foi a solução que você deu? Compartilha pra gente! :)

**Comment by yuricamara** [comment]
Eu trabalhei isso com [Node fs](https://nodejs.org/api/fs.html).
Pra incluir o code num pipe eu usei o [gulp-callback](https://www.npmjs.com/package/gulp-callback)

**Comment by felipefialho** [comment]
Valeu @yuricamara ;)
